adal/user_realm.py

```diff
@@ -131,6 +132,7 @@ def _parse_discovery_response(self, body):
             self.federation_protocol = protocol
             self.federation_metadata_url = response['federation_metadata_url']
             self.federation_active_auth_url = response['federation_active_auth_url']
+            self.cloud_audience_urn = response.get('cloud_audience_urn', "urn:federation:MicrosoftOnline")

```

Comment:
Confirmed that this would be the same approach as what ADAL C++ does for years now.

[4:21 PM] Petar Dimov
So what ADAL Cpp does is. If we receive it from the userrealm request, we will use what we receive. Otherwise we default to urn:federation:MicrosoftOnline

​[4:30 PM] Abhidnya Patil
    Cpp confirmed they pull it from userrealm response, Travis : https://github.com/AzureAD/azure-activedirectory-library-for-cpp/blob/dev/src/cpp/src/lib/windows/UserRealm.cpp#L233